nncf/quantization/algorithms/min_max/utils.py

```diff
@@ -11,37 +11,56 @@
  limitations under the License.
 """
 
-from typing import Union, List, Optional
-
+from typing import Tuple, List, Optional
+from dataclasses import dataclass
 import numpy as np
-
 from nncf.common.quantization.structs import QuantizerConfig
 from nncf.common.quantization.structs import QuantizationMode
 from nncf.common.tensor_statistics.statistics import MinMaxTensorStatistic
 
 
+@dataclass
 class QuantizerLayerParameters:
     """
-    Class handles Quantizer/Dequantizer layer attributes.
+    Class handles Quantizer layer attributes.
+    :param scale: Quantizer scale.
+    :param zero_point: Quantizer zero point.
+    :param mode: Quantizer mode. Could be Symmetric or Asymmetric.
+    :param axis: Axis for per-channel quantization. Should be none in case of per-tensor.
+    :param tensor_type: Signed or Unsigned tensor type.
     """
+    scale: List[float]
+    zero_point: List[int]
+    mode: QuantizationMode
+    axis: Optional[int] = None
+    tensor_type: Optional[np.dtype] = None
 
-    def __init__(self, scale: List[float], zero_point: List[int], mode: QuantizationMode, axis: Optional[int]):
-        self.scale = scale
-        self.zero_point = zero_point
-        self.mode = mode
-        self.axis = axis
 
-
-def calculate_scale_level(max_val: Union[float, np.ndarray], min_val: Union[float, np.ndarray],
-                          num_bits: int,
-                          mode: QuantizationMode) -> Union[float, np.ndarray]:
+def calculate_scale_level(max_val: np.ndarray, min_val: np.ndarray, level_low: int, level_high: int,
+                          mode: QuantizationMode) -> Tuple[np.ndarray, np.ndarray]:
     """
     Calculates Quantizer/Dequantizer layer scale level.
+    Returns scale and zero_point values foe the quantizer.
+
+    :param max_val: The maximum value of the input tensor.
+    :param min_val: The minimum value of the input tensor.
+    :param level_low: The minimum level of quantizer quants.
+    :param level_high: The maximum level of quantizer quants.
+    :param mode: Symmetric or Asymmetric mode.
+    :return: Scale and Zero point values.
+
     """
     if mode == QuantizationMode.SYMMETRIC:
         input_abs_max = np.maximum(np.abs(max_val), np.abs(min_val))
-        return input_abs_max / ((2 ** num_bits - 1) / 2)
-    return (max_val - min_val) / 2 ** num_bits
+        max_val = input_abs_max
+        min_val = -input_abs_max
+    scale = (max_val - min_val) / float(level_high - level_low)
+    if mode == QuantizationMode.SYMMETRIC:
+        zero_point = np.zeros_like(scale).astype(np.int64)
+    else:
+        zero_point = np.round(level_low - min_val / scale).astype(np.int64)
```

Comment:
The reference implementation:
https://github.com/tensorflow/tensorflow/blob/6642956b198ca002fc8a08458f1c14274be41915/tensorflow/core/kernels/fake_quant_ops_functor.h#L41

Reply:
Updated formula